fedimint-server/src/consensus/engine.rs

```diff
@@ -804,3 +1080,44 @@ pub async fn get_finished_session_count_static(dbtx: &mut DatabaseTransaction<'_
         .await
         .map_or(0, |entry| (entry.0 .0) + 1)
 }
+
+/// In memory copy of all the current consensus versions that are maintained in
+/// the DB.
+///
+/// To avoid reading from the database all the time, invalidation is supported.
+/// Any time consensus item that could have changed it is processed, the whole
+/// thing will get invalidated and reloaded at the beginning for processing new
+/// consensus item.
+///
+/// Create by using [`ConsensusEngine::load_consensus_version_cache`]
+#[derive(Debug)]
+struct ConsensusVersionCache {
+    // Currently nothing is using it, but in the future we can pass these
+    // to modules.
+    #[allow(unused)]
+    versions: BTreeMap<Option<ModuleInstanceId>, ConsensusVersionValue>,
+    invalidated: bool,
+}
```

Comment:
I thought that invalidatioon here is a simplification. There's only one way and place that can change a version - when the consensus version vote is processed. A write-through cache sounds roughly equivalent to me, except it would probably be a field on a struct, and not a variable.